config/main.go

```diff
@@ -0,0 +1,235 @@
+	// TODO: This needs to be manually overridden in the config to point to this
+	// repo. Eventually, I'll implement package downloads from github releases.
+	packageHome := getEnv("DOSAPP_PACKAGE_HOME", filepath.Join(stateHome, "packages"))
+
+	downloadHome := getEnv("DOSAPP_DOWNLOAD_HOME", filepath.Join(cacheHome, "downloads"))
+	pager := getEnv("PAGER", "cat")
+	diskA := getEnv("DOSAPP_DISK_A", filepath.Join(os.Getenv("HOME"), "Documents"))
+	diskB := getEnv("DOSAPP_DISK_B", "")
+	diskC := getEnv("DOSAPP_DISK_C", filepath.Join(os.Getenv("HOME"), "dosapp", "c"))
+
+	return Config{
+		mustExpandUser(root),
+		mustExpandUser(configHome),
+		logLevel,
+		mustExpandUser(dosBoxBin),
+		mustExpandUser(sevenZipBin),
+		mustExpandUser(dataHome),
+		mustExpandUser(stateHome),
+		mustExpandUser(cacheHome),
+		mustExpandUser(diskHome),
+		mustExpandUser(linkHome),
+		mustExpandUser(packageHome),
+		mustExpandUser(downloadHome),
+		mustExpandUser(pager),
+		mustExpandUser(diskA),
+		mustExpandUser(diskB),
+		mustExpandUser(diskC),
+	}
+}
+
+func editConfig(file string) error {
+	editor := os.Getenv("EDITOR")
+
+	if editor == "" {
+		return errors.New("No editor specified.")
+	}
+
+	cmd := exec.Command(editor, file)
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	return cmd.Run()
+}
+
+func (conf Config) Environ() []string {
+	env := []string{
+		"DOSAPP_ROOT=" + conf.Root,
+		"DOSAPP_CONFIG_HOME=" + conf.ConfigHome,
+		"DOSAPP_LOG_LEVEL=" + conf.LogLevel,
+		"DOSAPP_DOSBOX_BIN=" + conf.DosBoxBin,
+		"DOSAPP_7Z_BIN=" + conf.SevenZipBin,
+		"DOSAPP_DATA_HOME=" + conf.DataHome,
+		"DOSAPP_STATE_HOME=" + conf.StateHome,
+		"DOSAPP_CACHE_HOME=" + conf.CacheHome,
+		"DOSAPP_DISK_HOME=" + conf.DiskHome,
+		"DOSAPP_LINK_HOME=" + conf.LinkHome,
+		"DOSAPP_PACKAGE_HOME=" + conf.PackageHome,
+		"DOSAPP_DOWNLOAD_HOME=" + conf.DownloadHome,
+		"PAGER=" + conf.Pager,
+		"DOSAPP_DISK_A=" + conf.DiskA,
+		"DOSAPP_DISK_B=" + conf.DiskB,
+		"DOSAPP_DISK_C=" + conf.DiskC,
+	}
+
+	return append(os.Environ(), env...)
+}
+
+func (conf Config) WriteEnvFile() error {
+	envPath := filepath.Join(conf.ConfigHome, "dosapp.env")
+	return os.WriteFile(envPath, envFile, 0644)
+}
+
+func (conf Config) EditEnvFile() error {
+	envPath := filepath.Join(conf.ConfigHome, "dosapp.env")
+	return editConfig(envPath)
+}
+
+func (conf Config) EnvFileExists() bool {
+	envPath := filepath.Join(conf.ConfigHome, "dosapp.env")
+	_, err := os.Stat(envPath)
+	return err == nil
+}
+
+func (conf Config) WriteTaskFile() error {
+	taskPath := filepath.Join(conf.ConfigHome, "Taskfile.yml")
+	return os.WriteFile(taskPath, taskFile, 0644)
+}
+
+func (conf Config) TaskFileExists() bool {
+	taskPath := filepath.Join(conf.ConfigHome, "Taskfile.yml")
+	_, err := os.Stat(taskPath)
+	return err == nil
+}
+
+func (conf Config) Refresh() error {
+	err := conf.WriteTaskFile()
+	if err != nil {
+		return err
+	}
+
+	return conf.Run("init")
+}
+
+func (conf Config) Run(args ...string) error {
```

Comment:
I'd almost rather this functionality was in `dosapp/task`, but that would require moving `Refresh` to `dosapp/task` to avoid circular imports... which I don't want to do.